[904] Enable testing component JS interactions using new with_rendered_component_path method (#1061)
ViewComponent/view_component: docs/CHANGELOG.md

## main

* Add render_in_browser method to enable testing JS interactions in components

    *Edwin Mak*

* Replace antipatterns section in the documentation with best practices.

    *Blake Williams*

> 

Author: ```suggestion
```

## main

> * Add visit_rendered_in_browser method to enable testing JS interactions in components

Reviewer: How about this naming?

```suggestion
* Add `visit_rendered_component_in_browser` method to enable testing JS interactions in components.
```

## main

* Add visit_rendered_in_browser method to enable testing JS interactions in components

    *Edwin Mak*

> * Fix bug where `with_collection_parameter` did not inherit from parent component.
> 

Author: ```suggestion
```

## main

> * Add with_rendered_component_path method to enable testing JS interactions in components

Reviewer: ```suggestion
* Add with_rendered_component_path method to enable testing JS interactions in components.
```

> * Add with_rendered_component_path method to enable testing JS interactions in components.

Reviewer: ```suggestion
* Add `with_rendered_component_path` method to enable testing JS interactions in components.
```

> * Add `with_rendered_component_path` method to enable testing JS interactions in components.

Reviewer: ```suggestion
* Add `with_rendered_component_path` for writing component system tests.
```

> * Add with_rendered_component_in_browser method to enable testing JS interactions in components

Reviewer: ```suggestion
* Add `with_rendered_component_path` method to enable testing JS interactions in components
```

I thought I made this suggestion already, but maybe I didn't 😅 I think we should make it clear that we are not providing a browser.

> * Add visit_rendered_component_in_browser method to enable testing JS interactions in components

Reviewer: I'll write this here and assume you'll handle the other changes, but I think `with_rendered_component_path` is a good name for this new API ❤️ 

```suggestion
* Add `with_rendered_component_path` method to enable testing components in browser tests.
```

    *Matthew Rider*

* Fix bug where `with_collection_parameter` didn't inherit from parent component.

> >>>>>>> cb5b2da1be2ad73d7ebd828bfbd016acd09c9f58

Author: ```suggestion
```
